test/miscellaneous_tests/symbolic_stoichiometry.jl

```diff
@@ -1,256 +1,247 @@
 ### Prepares Tests ###
 
 # Fetch packages.
-using Catalyst, JumpProcesses, LinearAlgebra, OrdinaryDiffEq, Test
+using Catalyst, JumpProcesses, OrdinaryDiffEq, StochasticDiffEq, Test
 
 # Sets stable rng number.
 using StableRNGs
 rng = StableRNG(12345)
+seed = rand(rng, 1:100)
 
 # Sets the default `t` to use.
 t = default_t()
 
-### Base Tests ###
+# Fetch test functions.
+include("../test_functions.jl")
 
-# Probably requires us to permit Int64 parameters for this to work. 
-@test_broken let
-    @parameters k α
-    @species A(t), B(t), C(t), D(t)
-    rxs = [Reaction(t * k, [A], [B], [2 * α^2], [k + α * C])
-           Reaction(1.0, [A, B], [C, D], [α, 2], [k, α])]
-    @named rs = ReactionSystem(rxs, t)
-    @test issetequal(unknowns(rs), [A, B, C, D])
-    @test issetequal(parameters(rs), [k, α])
-    osys = convert(ODESystem, rs)
 
-    g = (k + α * C)
-    rs2 = @reaction_network rs begin
-        t * k, 2 * α^2 * A --> $g * B
-        1.0, α * A + 2 * B --> k * C + α * D
+### Base Tests ###
+
+# Checks that systems with symbolic stoichiometries, created using different approaches, are identical.
+let 
+    @parameters p k d n1 n2 n3
+    @species X(t) Y(t)
+    rxs1 = [
+        Reaction(p, nothing, [X], nothing, [n1])
+        Reaction(k, [X], [Y], [n2], [n3])
+        Reaction(d, [Y], nothing)
+    ]
+    rs1 = ReactionSystem(rxs1, t; name = :rs)
+
+    rxs2 = [
+        @reaction p, 0 --> n1*X
+        @reaction k, n2*X --> n3*Y
+        @reaction d, Y --> 0
+    ]
+    rs2 = ReactionSystem(rxs2, t; name = :rs)
+
+    rs3 = @reaction_network rs begin
+        p, 0 --> n1*X
+        k, n2*X --> n3*Y
+        d, Y --> 0
     end
-    @test rs2 == rs
 
-    rxs2 = [(@reaction t * k, 2 * α^2 * A --> $g * B),
-        (@reaction 1.0, α * A + 2 * B --> k * C + α * D)]
-    rs3 = ReactionSystem(rxs2, t; name = :rs)
-    @test rs3 == rs
+    @test rs1 == rs2 == rs3
+    @test issetequal(unknowns(rs1), [X, Y])
+    @test issetequal(parameters(rs1), [p, k, d, n1, n2, n3])
+end
 
-    u0map = [A => 3.0, B => 2.0, C => 3.0, D => 1.5]
-    pmap = (k => 2.5, α => 2)
-    tspan = (0.0, 5.0)
-    oprob = ODEProblem(osys, u0map, tspan, pmap)
-    # This is a hack because of https://github.com/SciML/ModelingToolkit.jl/issues/1475
-    oprob = remake(oprob, p = Tuple(pv[2] for pv in pmap))
-    du1 = zeros(size(oprob.u0))
-    oprob.f(du1, oprob.u0, oprob.p, 1.5)
+# Declares a network, parameter values, and initial conditions, to be used for the next couple of tests.
+begin
+    # Prepares parameters and species and their values.
+    @parameters k α::Int64
+    @species A(t), B(t), C(t), D(t)
+    u0_1 = [A => 3.0, B => 2.0, C => 3.0, D => 5.0]
+    ps_1 = [k => 5.0, α => 2.0]
```

Comment:
No, this seems to be a thing now. For e.g. `Int` they just have to be of a int form, and are then converted, so this was intended to be part of the tests.

Reply:
What do you mean? You are saying that MTK now converts parameters declared as integer to floating point and explicitly casts them back to integers within the generated functions? That doesn't sound at all like how MTKParameters is supposed to work.

Reply:
Can check with them, but something like that, yes. If a parameter is declared as an integer an its input can be interpreted is an integer then it is done so (basically what happens in the code above).

I can ask them if this is working as intended though.

Reply:
@AayushSabharwal is what @TorkelE is saying correct? Declaring a parameter as integer results in the generated MTKParameter object storing it as floating point internally and then calling `Int(p)` on it within generated functions? I would have thought that the whole point of `MTKParameter` is to allow us to specify it as an integer within the parameter map based on it being declared as a symbolic integer, and to then propagate this everywhere without having to cast back and forth to a float.

Reply:
No. Parameters declared as integers (`@parameters p::Int`) are stored as integers. They can be initialized with a floating point value that represents an integer because the initialization will call `convert(Int, val)`. Using a `Vector{Pair}` can lead to issues, since `[p1 => 1.5, p2 => 1]` is a `Vector{Pair{..., Float64}}` (the int literal is converted to `Float64` similar to how `[1.5, 1, 2]` is a `Vector{Float64}`)

Reply:
What’s the recommended input format to use then with mixed types? Just declare the vector to have type Any? Use a tuple?

Reply:
@TorkelE I think we can just use tuples as we used to do for mixed maps. At least `DiscreteProblem` seems to work with them ok it seems.

Reply:
I can use mixed Tuples here. We are testing that mixed Tuples/vectors works identically in the other test anyway, so which we use here should not matter.

Reply:
I'd recommend a `Dict`. Everything is eventually converted to a `Dict` anyway

Reply:
@TorkelE either Tuple or Dict is fine with me. I guess with a Dict you'll have to think about how to specify ordering for our custom ODE derivative functions.

Reply:
I now use Tuples here, but this does not really have any impact on the tests